### ISSUE TITLE
Build packages for main linux distributions, including systemd startup scripts

### ISSUE THREAD

**Issue by tlaurion** [etesync/etesync-dav#121]

**Comment by tasn** [comment]
As you hinted, there is a plugin in the works for Thunderbird, and potentially also for KDE and GNOME (Evolution) during GSoC: https://blog.etesync.com/summer-internship-opportunities-etesync-projects-in-both-gsoc-and-outreachy/

As for the packages, I completely agree. It's currently packaged for Arch (in AUR) and if memory serves also NixOS. There are maybe also packages for fedora and centos. I don't remember.
Either way, packaging, unfortunately, is done by the distributions maintainers, and not externally, so in order to get it in distros, you would probably want to open a ticket in your respective distro's issue tracker.

As for systemd startups scripts: we have them in the repo, so it'd be very easy for the package to install them too.

**Comment by tlaurion** [comment]
Test packages are loves with spec files, so that external builders can easily pickup and distribution maintainers just add those spec files in their builders and sign packages!

**Comment by tasn** [comment]
Yeah, we would love having those (in addition to the ones we have), do you know how to package for any distro? PRs would be highly welcome.

**Comment by tlaurion** [comment]
Would suggest to put a tag: help needed to help you get people knowledgeable do those needed tasks that are not planned right now from core devels of this project!

**Comment by tasn** [comment]
@tlaurion, thanks, added the label too. I agree about the organic comment, there are already a few packages, we just need to make sure they are listed in the readme.

**Comment by daftaupe** [comment]
> Edit: @daftaupe, could you please elaborate on the state of packaging of etesync-dav for fedora/centos? Thanks!

@tasn I haven't packaged etesync-dav (only the etesync server so far), as I'm not using it myself for now.

I can try to package if people are interested.

**Comment by tasn** [comment]
Ah, I must have misremembered. If you can take a look, that would be great! I guess it all depends on how many of the deps are already packaged for the respective distros...

**Comment by tlaurion** [comment]
@daftaupe ?

**Comment by daftaupe** [comment]
@tlaurion well it's hard to find the time at the moment, but I'll look into that sooner or later :)

**Comment by tlaurion** [comment]
@daftaupe ping!

**Comment by BeatLink** [comment]
Id be so happy to see a deb package. You can release deb packages on github releases or even better, create your own PPA. In both cases you dont need to wait on maintainers to integrate it first.

**Comment by tasn** [comment]
I'd also be super happy, and I'd even happily build the dep packages for each release, I just need someone to actually implement the deb package (so I can build it).

**Comment by tlaurion** [comment]
Meanwhile, this tool is unfortunately not useable by a team and require too much support. I'll move away of it.

**Comment by tasn** [comment]
@tlaurion, what do you mean by "require too much support"? I agree we should add more packages though.
Not sure what you mean by "the real issue should be resolved upstream" as NextCloud can't and could never support end-to-end encryption.

**Comment by tlaurion** [comment]
@tasn : 

> @tlaurion, what do you mean by "require too much support"? I agree we should add more packages though.

Deploying and maintaining etesync is not possible until tools required to use such tools exist and is automated on desktop levels, where most of task creating and time management related tasks are done from desktop, not smartphones.

> Not sure what you mean by "the real issue should be resolved upstream" as NextCloud can't and could never support end-to-end encryption.

Well, "can't" and "could never" are pretty strong claims, which were technical challenges resolves by etesync that sould be also resolved upstream with enough awareness and will.

Most of NextCloud users are not even aware that calendar and tasks are not encrypted at all. If that awareness reached users and users needs met investment, that technical problem would be tackled, and the ideas behing etesync could be picked up by upstream project. 

Data stored on remote servers could be encrypted just like etesync did it, with a client-side encryption mechanism being implemented to do the actual encryption/decryption upon stored data across users knowing the shared secret.

**Comment by tasn** [comment]
> Deploying and maintaining EteSync is not possible until tools required to use such tools exist and installation and configuration becomes automated on desktop level just like it was on smartphones; desktop being where most of tasks related to time management really happens; not smartphones where those are calendar and tasks users and executors. In a collaborator environment where multiple desktops end users participate in the creation of tasks and events content for others, the actual smartphone-only-easy-deployment-with-self-enrollment is more then problematic.

I agree. We would love to have more of it. Though we are also focusing on making it so most people won't even need etesync-dav. Take a look at:
https://blog.etesync.com/gnome-and-kde-etesync-projects-accepted-to-gsoc/
https://blog.etesync.com/etesync-thunderbird-add-on/

> Well, "can't" and "could never" are pretty strong claims, which were technical challenges resolved by etesync that should also be resolved upstream by NextCloud with enough awareness and will.
> 
> Most of NextCloud users are not even aware that calendar and tasks are not encrypted at all. If that awareness reached users and users needs met investment, that technical problem would be tackled, and the ideas behing etesync could be picked up by upstream project.
> 
> Data stored on remote servers could be encrypted just like etesync did it, with a client-side encryption mechanism being implemented to do the actual encryption/decryption upon stored data across users knowing the shared secret.

Let me clarify what I meant: of course NextCloud could mimic EteSync. Heck, they could just take EteSync itself and just run it inside NextCloud, so of course this is not what I meant.

What I meant is that NextCloud is a DAV server. They are all about compatibility with existing implementations and following this standard, and thus are very unlikely to implement end-to-end encryption until there's a widely adopted standard.

**Comment by tlaurion** [comment]
@daftaupe : how is the packaging going? (debian-10/fedora-32)?
